include/functions.h

```diff
@@ -1266,35 +1266,33 @@ void SoundSource_PlaySfxEachFrameAtFixedWorldPos(PlayState* play, Vec3f* worldPo
 u16 QuestHint_GetTatlTextId(PlayState* play);
 
 u16 Text_GetFaceReaction(PlayState* play, u32 reactionSet);
-void EnvFlags_UnsetAll(PlayState* play);
-void EnvFlags_Set(PlayState* play, s16 flag);
-void EnvFlags_Unset(PlayState* play, s16 flag);
-s32 EnvFlags_Get(PlayState* play, s16 flag);
-s16 func_800F1460(s16 param_1);
-ActorCutscene* ActorCutscene_GetCutsceneImpl(s16 index);
-void ActorCutscene_Init(PlayState* play, ActorCutscene* cutscenes, s32 num);
-void func_800F15D8(Camera* camera);
-// void ActorCutscene_ClearWaiting(void);
-// void ActorCutscene_ClearNextCutscenes(void);
-// void ActorCutscene_MarkNextCutscenes(void);
-// void ActorCutscene_End(void);
-// void ActorCutscene_Update(void);
-void ActorCutscene_SetIntentToPlay(s16 index);
-s16 ActorCutscene_GetCanPlayNext(s16 index);
-s16 ActorCutscene_StartAndSetUnkLinkFields(s16 index, Actor* actor);
-s16 ActorCutscene_StartAndSetFlag(s16 index, Actor* actor);
-s16 ActorCutscene_Start(s16 index, Actor* actor);
-s16 ActorCutscene_Stop(s16 index);
-s16 ActorCutscene_GetCurrentIndex(void);
-ActorCutscene* ActorCutscene_GetCutscene(s16 index);
-s16 ActorCutscene_GetAdditionalCutscene(s16 index);
-s16 ActorCutscene_GetLength(s16 index);
-s32 func_800F2138(s32 arg0);
-s32 func_800F2178(s16 arg0);
-s16 ActorCutscene_GetCurrentSubCamId(s16 index);
-s16 func_800F21CC(void);
-s32 func_800F22C4(s16 param_1, Actor* actor);
-void ActorCutscene_SetReturnCamera(s16 index);
+
+void CutsceneFlags_UnsetAll(PlayState* play);
+void CutsceneFlags_Set(PlayState* play, s16 flag);
+void CutsceneFlags_Unset(PlayState* play, s16 flag);
+s32 CutsceneFlags_Get(PlayState* play, s16 flag);
+
+void ActorCutscene_Init(PlayState* play, ActorCutscene* actorCutsceneList, s16 num);
+void ActorCutscene_StoreCamera(Camera* camera);
+void ActorCutscene_ClearWaiting(void);
+s16 ActorCutscene_Update(void);
+void ActorCutscene_SetIntentToPlay(s16 csId);
```

Comment:
This adds the cutscene id to the waiting cutscenes. This name feels too verbose, I think something like `ActorCutscene_Queue` gets the idea across similarly, it's being added to the waiting cutscene list which acts as a priority queue for the cutscenes

Reply:
I think it's better than what it is currently, but I still want to give this some thought. Changing it for now, but going to leave it unresolved until I've thought about it more

Reply:
Okay, so `ActorCutscene_SetIntentToPlay`/`ActorCutscene_GetCanPlayNext` (or `CutsceneManager_Queue`/`CutsceneManager_IsNext`) along with `sNextCutsceneList`/`sWaitingCutsceneList` is really just a system of convenience for actors to coordinate their cutscenes, but nothing about those functions and data actually prevents or changes anything about `CutsceneManager_Start`.

i.e.

it's expected for the actor to follow the style of:
```c
if (CutsceneManager_IsNext(csId)) {
    CutsceneManager_Start(csId, &this->actor);
    ...
else {
    CutsceneManager_Queue(csId);
    ...
}
```

But it doesn't have to follow that system at all, it can just go straight to 
```c
CutsceneManager_Start(csId, &this->actor);
```
and play just fine. The only issue is when two cutscenes try to play at once. Calling `CutsceneManager_Start` when there's another cutscene playing will just simply reject that cutscene. So having this `CutsceneManager_Queue`/`CutsceneManager_IsNext` system prevents that.

So addressing a few comments at once: having a priority of `-1` means that if a cutscene is "queued", then `CutsceneManager_IsNext` will always return true. i.e. it's no different than just calling `CutsceneManager_Start` directly.

Of course, what `sNextCutsceneList` stores is true for all cutscenes with priority `-1` and the single cutscene that has the highest priority (via the lowest value).

So because of that, I'm wondering about the name `CutsceneManager_Queue` because it's almost like it's a ``CutsceneManager_QueueIntentToPlay`, i.e. it's not telling `CutsceneManager` to play it, it's more indicating to other cutscenes that this cutscene intends to play, and that it should not allow other cutscenes to play with a lower priority. And also with `CutsceneManager_IsNext`, many cutscenes can return true for that given cutscenes with a -1 priority. It's hard to articulate my thoughts, I'll keep thinking about these names